workflow-service/src/main/java/com/redhat/parodos/workflow/definition/service/WorkFlowDefinitionServiceImpl.java

```diff
@@ -82,15 +82,28 @@ public WorkFlowDefinitionServiceImpl(WorkFlowDefinitionRepository workFlowDefini
 		this.modelMapper = modelMapper;
 	}
 
+	private HashMap<String, Map<String, String>> convertWorkFlowParameters(List<WorkFlowParameter> workFlowParameters) {
+		HashMap<String, Map<String, String>> result = new HashMap<>();
+		for (WorkFlowParameter workFlowParameter : workFlowParameters) {
+			if (workFlowParameter == null)
+				continue;
+
+			result.put(workFlowParameter.getKey(), workFlowParameter.getAsJsonSchema());
+		}
+		return result;
+	}
+
 	@Override
 	public WorkFlowDefinitionResponseDTO save(String workFlowName, WorkFlowType workFlowType,
 			List<WorkFlowParameter> workFlowParameters, List<Work> works,
 			WorkFlowProcessingType workFlowProcessingType) {
+
 		// set and save workflow definition
-		WorkFlowDefinition workFlowDefinition = workFlowDefinitionRepository
-				.save(WorkFlowDefinition.builder().name(workFlowName).type(workFlowType).createDate(new Date())
-						.parameters(WorkFlowDTOUtil.writeObjectValueAsString(workFlowParameters)).modifyDate(new Date())
-						.numberOfWorks(works.size()).processingType(workFlowProcessingType.name()).build());
+		WorkFlowDefinition workFlowDefinition = workFlowDefinitionRepository.save(WorkFlowDefinition.builder()
+				.name(workFlowName).type(workFlowType).createDate(new Date())
+				.parameters(WorkFlowDTOUtil.writeObjectValueAsString(convertWorkFlowParameters(workFlowParameters)))
+				.modifyDate(new Date()).numberOfWorks(works.size()).processingType(workFlowProcessingType.name())
+				.build());
```

Comment:
Difficult to read: can you fix it?

This also apply to other similar code blocks.